### PR TITLE
derivation of energy formula

### PR DESCRIPTION
**Pull request by cslade1970**

### CODE REVIEW

File: symplyphysics/laws/electricity/power_from_energy_time.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 from symplyphysics import (
-    symbols, Eq, pretty, solve, Quantity, units,
+    symbols, Eq, Function, simplify, pretty, solve, Quantity, units,
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/power_from_energy_time.py:2]: Function is not used.

File: symplyphysics/laws/electricity/power_from_energy_time.py [hunk 2]
```diff
@@ -12,6 +13,17 @@
 # TODO: this law should be derived from definition power_is_energy_derivative
 power, energy, time = symbols('power energy time')
 law = Eq(power, energy / time)
+constant_1, constant_2 = symbols('constant_1 constant_2')
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/power_from_energy_time.py:16]: Separate derivation from the law and add explanations.

Comment (blackyblack) [inline symplyphysics/laws/electricity/power_from_energy_time.py:16]: Might be a good idea to hint what are these constants. I guess constant_1 is constant_power and constant_2 is initial energy.

File: symplyphysics/laws/electricity/power_from_energy_time.py [hunk 3]
```diff
@@ -12,6 +13,17 @@
...
+linear_energy = constant_1 * power_derivative.time +constant_2
+definition_applied = power_derivative.definition.rhs.subs(power_derivative.energy_function(time), linear_energy)
+linear_power = definition_applied.doit()
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/power_from_energy_time.py:19]: It's not linear power. More like constant power from linear energy.

File: symplyphysics/laws/electricity/power_from_energy_time.py [hunk 4]
```diff
@@ -12,6 +13,17 @@
 # TODO: this law should be derived from definition power_is_energy_derivative
 power, energy, time = symbols('power energy time')
 law = Eq(power, energy / time)
+constant_1, constant_2 = symbols('constant_1 constant_2')
+linear_energy = constant_1 * power_derivative.time +constant_2
+definition_applied = power_derivative.definition.rhs.subs(power_derivative.energy_function(time), linear_energy)
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/power_from_energy_time.py:18]: energy_function is a function of `power_derivative.time`, not `time`.

Comment (blackyblack) [inline symplyphysics/laws/electricity/power_from_energy_time.py:18]: I think it is a good idea to use local `time` variable. Code will be shorter and you subs it down the code anyway.

File: symplyphysics/laws/electricity/power_from_energy_time.py [hunk 5]
```diff
@@ -12,6 +13,17 @@
...
+linear_power = definition_applied.doit()
+definition_energy = linear_energy.subs({constant_1: power, constant_2: 0, power_derivative.time: time})
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/power_from_energy_time.py:20]: Explain, why constant_2 is set to 0.

Comment (blackyblack) [inline symplyphysics/laws/electricity/power_from_energy_time.py:20]: How do you know that constant_1 should be replaced with power?

File: symplyphysics/laws/electricity/power_from_energy_time.py [hunk 6]
```diff
@@ -12,6 +13,17 @@
...
+dsolved = definition_energy.doit()
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/power_from_energy_time.py:21]: This is not a differential equation so you neither need `doit()` nor dsolved.

**Review by cslade1970 — COMMENTED** [review]
reviewed

**Pull request merged**